packages/android-client/FishjamClient/src/main/java/com/fishjamcloud/client/CommandsQueue.kt

```diff
@@ -0,0 +1,47 @@
+package com.fishjamcloud.client
+
+import kotlinx.coroutines.Job
+
+internal class CommandsQueue {
+  private var commandsQueue: ArrayDeque<Command> = ArrayDeque()
+  var clientState: ClientState = ClientState.CREATED
+
+  fun addCommand(command: Command): Job {
+    commandsQueue.add(command)
+    if (commandsQueue.size == 1) {
+      command.command()
+    }
+    return command.job
+  }
+
+  fun finishCommand() {
+    val command = commandsQueue.first()
+    val job = command.job
+    commandsQueue.removeFirst()
+    if (command.clientStateAfterCommand != null) {
+      clientState = command.clientStateAfterCommand
+    }
+    job.complete()
+    // TODO: make it iterative, not recursive?
```

Comment:
Could we have any task for this TODO with some explanation?

Reply:
https://linear.app/swmansion/issue/FCE-303/make-commands-queue-iterative-instead-of-recursive

this is very low priority though